File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        
        ans = 0

        a = 0

        i = 0
        for j in range(len(s)):
            a =a + abs(ord(s[j]) - ord(t[j]))
            
            while a>maxCost:
                a = a - abs(ord(s[i]) - ord(t[i]))
                i+=1
            ans = max(ans,j-i+1)
        return ans

File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int ans = 0;
        int a = 0;
        int i = 0;
        for(int j = 0 ;j<s.size();j++){
            a = a + abs(s[j] - t[j]);
            while(a>maxCost){
                a = a - abs(s[i] - t[i]);
                i+=1;
            }
            ans = max(ans,j-i+1);
        }   
        return ans;
    }
};